Framework/API/src/IFunction.cpp

```diff
@@ -108,12 +109,16 @@ void IFunction::functionDeriv(const FunctionDomain &domain,
  * with this reference: a tie or a constraint.
  * @return newly ties parameters
  */
-ParameterTie *IFunction::tie(const std::string &parName,
-                             const std::string &expr, bool isDefault) {
-  auto ti = new ParameterTie(this, parName, expr, isDefault);
-  addTie(ti);
+void IFunction::tie(const std::string &parName, const std::string &expr,
+                    bool isDefault) {
+  auto ti = std::make_unique<ParameterTie>(this, parName, expr, isDefault);
```

Comment:
I think you'll need to use `Kernel::make_unique` here. `std::make_unique` is not yet available on gcc 4.8 which is being used on RHEL6.